### PR TITLE
feat: usermanager

### PR DESCRIPTION
**Pull request by dni**
add usermanager to admin_ui. show user accounts and wallets with transaction count and balance

screenshot-1702563067

- adds `manage user section`
- move topup to users
- enable / disable admins

### CODE REVIEW

**Comment by arcbtc** [comment]
A user could also spam and make 1000s of wallets, so a delete all button for wallets would be good.

File: lnbits/core/views/user_api.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+from lnbits.core.services import update_wallet_balance
+from lnbits.db import Filters, Page
+from lnbits.decorators import check_admin, check_super_user, parse_filters
+from lnbits.settings import EditableSettings, settings
+
+users_router = APIRouter(prefix="/users/api/v1", dependencies=[Depends(check_admin)])
+
+
+@users_router.get("/user/")
+async def api_users_get_users(
```
Comment (motorina0) [inline lnbits/core/views/user_api.py:26]:
```suggestion
async def api_get_users(
```

File: lnbits/core/views/user_api.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            detail="Cannot delete super user.",
+        )
+
+    if user_id in settings.lnbits_admin_users and not user.super_user:
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="Only super_user can delete admin user.",
+        )
+
+    await delete_account(user_id)
```
Comment (motorina0) [inline lnbits/core/views/user_api.py:69]: try-catch

File: lnbits/core/views/user_api.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail=f"Could not fetch users. {exc}",
+        ) from exc
+
+
+@users_router.delete("/user/{user_id}", status_code=HTTPStatus.OK)
+async def api_users_delete_user(
+    user_id: str, user: User = Depends(check_admin)
+) -> None:
+    wallets = await get_wallets(user_id)
```
Comment (motorina0) [inline lnbits/core/views/user_api.py:51]: try-catch

File: lnbits/core/views/user_api.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="Cannot change super user.",
+        )
+    if user_id in settings.lnbits_admin_users:
+        settings.lnbits_admin_users.remove(user_id)
+    else:
+        settings.lnbits_admin_users.append(user_id)
+    update_settings = EditableSettings(lnbits_admin_users=settings.lnbits_admin_users)
+    await update_admin_settings(update_settings)
```
Comment (motorina0) [inline lnbits/core/views/user_api.py:79]: try-catch

File: lnbits/core/views/user_api.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+async def api_users_delete_user_wallet(user_id: str, wallet: str) -> None:
+    wal = await get_wallet(wallet)
+    if not wal:
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="Wallet does not exist.",
+        )
+    if wal.deleted:
+        await force_delete_wallet(wallet)
+    await delete_wallet(user_id=user_id, wallet_id=wallet)
```
Comment (motorina0) [inline lnbits/core/views/user_api.py:120]: should check if its not `super_user` or another admin user?

**Review by motorina0 — APPROVED** [review]

**Pull request merged**